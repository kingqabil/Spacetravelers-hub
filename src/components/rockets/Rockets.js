import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Rockets from './Rocket';
import { fetchRockets } from '../../redux/rockets/rockets';

const DisplayRockets = () => {
  const dispatch = useDispatch();

  const rocketsStore = useSelector((store) => store.rocketsReducer.rockets);

  useEffect(() => {
    if (!rocketsStore.length) {
      fetchRockets(dispatch);
    }
  }, []);

  return (
    <div className="Rockets">

      {
        rocketsStore.map((rockets) => (
          <Rockets
            key={rockets.id}
            id={rockets.id}
            flickr_images={rockets.flickr_images}
            rocket_name={rockets.rocket_name}
            description={rockets.description}
            reserved={rockets.reserved}
          />
        ))
      }
    </div>
  );
};

export default DisplayRockets;
