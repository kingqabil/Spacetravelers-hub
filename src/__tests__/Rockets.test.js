import React from 'react';
import { Provider } from 'react-redux';
import Rockets from '../components/rockets/Rockets';
import store from '../redux/configureStore';
import {render} from '@testing-library/react'

const ReactTestRenderer = require('react-test-renderer');


describe('Rockets component tests', () => {

	it('renders correctly', () => {
		const tree = ReactTestRenderer
			.create(
				<Provider store={store}><Rockets /></Provider>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
  
	it('has a text in the DOM', () => {
		const { container } = render(<Provider store={store}><Rockets /></Provider>)
		expect(container.getElementsByClassName('rockets-list').length).toBe(1);
})
})
