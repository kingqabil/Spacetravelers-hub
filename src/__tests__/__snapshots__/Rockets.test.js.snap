// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component tests renders correctly 1`] = `
<div
  className="rockets-list"
/>
`;
